src/main/java/commands/SearchCommand.java

```diff
@@ -0,0 +1,62 @@
+package commands;
+
+import storage.Storage;
+import task.Deadline;
+import task.Event;
+import task.Task;
+import tasklist.TaskList;
+import ui.Ui;
+
+import java.time.LocalDate;
+
+/**
+ * Search Task by date and list Task.
+ */
+public class SearchCommand extends Command {
+
+    private LocalDate date;
+
+    /**
+     * Constructor for SearchCommand.
+     *
+     * @param date This is the date of Tasks we are searching for.
+     */
+    public SearchCommand(LocalDate date) {
+        this.date = date;
+    }
+
+    /**
+     * Execute the SearchCommand. It searches for Tasks of an user input date.
+     *
+     * @param tasks This is the TaskList where the Task is stored.
+     * @param ui This is to interact with the user interface, printing message of Task having the required date.
+     * @param storage Unused.
+     */
+    @Override
+    public String execute(TaskList tasks, Ui ui, Storage storage) {
+        String msg = "Here are the tasks on " + this.date + ":";
+        for (Task t : tasks.getTasks()) {
+            if (t instanceof Deadline) {
```

Comment:
In future, one possible improvement could be to extend Tasks with dates from some abstract class/interface to perform this search, which would avoid multiple if/else statements and follow Open-Closed Principle more closely (more Task types can be added without modifying other classes)